Add Antrea as Network Policy Provider (#24992)
kubernetes/website: content/en/docs/tasks/administer-cluster/network-policy-provider/antrea-network-policy.md

---
title: Use Antrea for NetworkPolicy
content_type: task
weight: 10
---

<!-- overview -->
This page shows how to create an Antrea cluster on Kubernetes.
For background on Project Antrea, read the [Introduction to Antrea](https://antrea.io/docs/).

## {{% heading "prerequisites" %}}

> You need a Kubernetes cluster created on premisses or in Public Cloud, and the kubectl command-line tool must be configured to communicate with your cluster.

Reviewer: ```suggestion
You need a Kubernetes cluster created on premises or in public cloud, and the kubectl command-line tool must be configured to communicate with your cluster.
```

<!-- steps -->

## Deploying Antrea with kubeadm

When using `kubeadm` to create the Kubernetes cluster, passing
`--pod-network-cidr=<CIDR Range for Pods>` to `kubeadm init` will enable

> `NodeIpamController`. Clusters created with kubeadm will always have

Reviewer: ```suggestion
`NodeIpamController`. Clusters created with kubeadm always have
```

`CNI` plugins enabled. Refer to

> [Creating a cluster with kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm)

Reviewer: ```suggestion
[Creating a cluster with kubeadm](/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm)
```

for more information about setting up a Kubernetes cluster with `kubeadm`.

To deploy the latest version of Antrea (built from the master branch), use the

> checked-in [deployment yaml](https://github.com/vmware-tanzu/antrea/blob/master/build/yamls/antrea.yml):

Reviewer: ```suggestion
checked-in [deployment manifest](https://github.com/vmware-tanzu/antrea/blob/master/build/yamls/antrea.yml):
```

```bash
kubectl apply -f https://raw.githubusercontent.com/vmware-tanzu/antrea/master/build/yamls/antrea.yml
```

## Deploying Antrea in Kind

To deploy Antrea in a [Kind](https://github.com/kubernetes-sigs/kind) cluster,

> please refer to this [guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/kind.md).

Reviewer: ```suggestion
refer to [Deploying Antrea on a Kind cluster](https://github.com/vmware-tanzu/antrea/blob/master/docs/kind.md).
```

## Deploying Antrea in EKS and GKE

Antrea can be deployed in NetworkPolicy only mode to an EKS cluster or a GKE
cluster, and enforce NetworkPolicies for the cluster.

* To deploy Antrea in an EKS cluster, please refer to [the EKS installation guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/eks-installation.md).

> * To deploy Antrea in a GKE cluster, please refer to [the GKE installation guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/gke-installation.md).

Reviewer: A task page should focus as much as possible on _a_ task. It's OK to focus on the kind, or minikube, or GKE story - but that focus _is_ important. I'd remove this text.

Author: @sftim thanks fo your help. This makes sense. I will try to reduce the content.

> ## Deploying Antrea with IPsec Encryption

Reviewer 2: i think it is worth while adding a reference to the Antrea on Windows config, which also provides a NetworkPolicy implementation for Windows

Antrea supports encrypting GRE tunnel traffic with IPsec. To deploy Antrea with

> IPsec encryption enabled, please refer to this [guide](https://github.com/vmware-tanzu/antrea/blob/master/docs/ipsec-tunnel.md).

Reviewer: ```suggestion
 IPsec encryption enabled, please refer to [IPsec Encryption of Tunnel Traffic with Antrea](https://github.com/vmware-tanzu/antrea/blob/master/docs/ipsec-tunnel.md).
```
